### PR TITLE
Add env variable NEXT_PUBLIC_INVITE_DISABLED to allow or not for team invite

### LINKED ISSUE #250: Allow team invites when public signup is disabled
At the moment you can set the environment variable  `NEXT_PUBLIC_SIGNUP_DISABLED=1` to disable the public signup. Unfortunately this also breaks the team invite functionality as users can't register after receiving an invite link via email.

To solve this add a new environment variable `NEXT_PUBLIC_INVITE_DISABLED=1` that explicitly disables the invite functionality together with the signup for invites.
And change the default behaviour that if `NEXT_PUBLIC_SIGNUP_DISABLED=1` signup for invites is still allowed.

### PR DESCRIPTION
**Pull request by niteshseram**
## What does this PR do?

Earlier, when this env variable `NEXT_PUBLIC_SIGNUP_DISABLED` is set to 1, the team invite is also blocked. So, added an env variable `NEXT_PUBLIC_INVITE_DISABLED` which will block the team's user invite flow to signup only when this `NEXT_PUBLIC_SIGNUP_DISABLED` is set to 1.

Fixes #250 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## Checklist

- [x] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Review by mattinannt — CHANGES_REQUESTED** [review]
@niteshseram thanks a lot for this contribution 🤗
I like how you got around the disabled login for invites. I just found the problem here that you can insert any invite token (e.g. `abcde`) and still create a new account. I think it would be great if we could check here if the invite token is valid.

Also missing are all the other restrictions that `NEXT_PUBLIC_INVITE_DISABLED: 1` brings. When this flag is set, invitations should be disabled:
- The endpoints for inviting new members should be disabled.
- The `Add Member` button in `/settings/members` should be hidden.

Can you please add these missing features?

**Comment by niteshseram** [comment]
@mattinannt Do you think we need to verify the token when the user goes to signup with some dummy token as well? Because we do verify the token when the user try to signup using that token and if it is invalid, it just throw you the error.

**Comment by mattinannt** [comment]
> @mattinannt Do you think we need to verify the token when the user goes to the signup page with some dummy token as well? Because we do verify the token when the user tries to signup using that token and if it is invalid, it just throws you the error.

Yes, you're right, I haven't seen the login fail with an invalid token. Then everything is ok in that regard 😊.

**Comment by niteshseram** [comment]
@mattinannt Request changes are done 🙌

**Review by mattinannt — CHANGES_REQUESTED** [review]
@chetan Thanks for the changes 😊 I just have one small change request to make the invite functionality work when `NEXT_PUBLIC_INVITE_DISABLED` is not set/undefined, but after that we are ready to merge 💪🎉.

File: apps/web/app/environments/[environmentId]/settings/members/EditMemberships.tsx [hunk 1]
```diff
@@ -77,13 +77,16 @@ export function EditMemberships({ environmentId }: EditMembershipsProps) {
...
         </Button>
-        <Button
-          variant="primary"
-          onClick={() => {
-            setAddMemberModalOpen(true);
-          }}>
-          Add Member
-        </Button>
+
+        {process.env.NEXT_PUBLIC_INVITE_DISABLED === '0' && (
```
[the remark this answers is not in the document]
Comment (niteshseram) [reply apps/web/app/environments/[environmentId]/settings/members/EditMemberships.tsx]: Thanks @mattinannt. Good point. Missed from my end.

**Review by mattinannt — APPROVED** [review]
Thanks for the changes 😊 Works great 💪
I just solved a merge conflict due to another PR I merged previously 😊
Thank you for the great work; I will merge the changes into main now 🎉

**Pull request merged**